src/Analyzers/CSharp/Tests/MisplacedUsingDirectives/MisplacedUsingDirectivesCodeFixProviderTests.cs

```diff
@@ -943,10 +943,10 @@ namespace TestNamespace
 }
 ";
 
-            var fixedTestCode = @"namespace TestNamespace
+            var fixedTestCode = @"// Some comment
+namespace TestNamespace
 {
 
-    // Some comment
     {|Warning:using System;|}
```

Comment:
is this change intentional?  it seems undesirable.

Reply:
The change was introduced in 7650e92d and it was intentional.

The `MisplacedUsingDirectivesCodeFixProvider` removes any leading banner from the `using` trivia before the move:

https://github.com/dotnet/roslyn/blob/672e56555642a433440f06289ecd93adfc193533/src/Analyzers/CSharp/CodeFixes/MisplacedUsingDirectives/MisplacedUsingDirectivesCodeFixProvider.cs#L123

Given that, any comment that is considered a banner should be left behind. In the past, it wasn't a requirement to place an empty newline between the header and the first using, so imho we need to treat `//Comment` as header and leave it behind (please note, that before this PR there were different implementations of "what is considered a file banner" which treated edge cases like this slightly different.

A related unresolved issue is described in https://github.com/dotnet/roslyn/pull/49440#issuecomment-733936704 for block comments.

Please note, that we must treat `//Comment` as a file banner because otherwise, we would break existing code with "Warning file banner missing".